### ISSUE TITLE
Lint errors should include TypeScript version

### ISSUE THREAD

**Issue by leonard-thieu** [Microsoft/dtslint#146]
```
Error: S:/Projects/_GitHub/leonard-thieu/DefinitelyTyped/types/jquery/jquery-tests.ts:8021:1
ERROR: 8021:1  expect  Expected an error on this line, but found none.
```
[Full log](https://github.com/Microsoft/dtslint/files/2478861/log.txt)

In this scenario, I _know_ it's a TypeScript 2.3 error because it's a weak type test but I don't think it'll always be so obvious. I think printing the TypeScript version that the rule failed with would be helpful in debugging test failures.
